src/hyperlight_host/src/hypervisor/dump_on_crash.rs

```diff
@@ -0,0 +1,47 @@
+use std::io::Write;
+
+use tempfile::NamedTempFile;
+
+use super::Hypervisor;
+use crate::{new_error, Result};
+
+/// Dump registers + memory regions + raw memory to a tempfile
+#[cfg(feature = "dump_on_crash")]
+pub(crate) fn dump_on_crash_to_tempfile(hv: &dyn Hypervisor) -> Result<()> {
+    let mem_regions = hv.get_memory_regions();
+    let mem_size = mem_regions
+        .iter()
+        .map(|region| region.host_region.len())
+        .sum();
+    let mem_start_addr = mem_regions[0].host_region.start as *const u8;
```

Comment:
It looks like an invariant here that the host regions all be contiguous, which is always the case now, but might not be in the future. Can we add a note on the definition of `get_memory_regions()` that this is an invariant imposed by this code, and/or change this code to not impose that invariant?

Reply:
fixed!